tests/kafkatest/services/verifiable_producer.py

```diff
@@ -145,6 +146,10 @@ def _worker(self, idx, node):
             producer_prop_file += "\nmax.in.flight.requests.per.connection=5\n"
             producer_prop_file += "\nretries=1000000\n"
             producer_prop_file += "\nenable.idempotence=true\n"
+        elif self.retries is not None:
+            self.logger.info("VerifiableProducer (index = %d) will use retries = %s", idx, self.retries)
+            producer_prop_file += "\nretries=%s\n" % self.retries
+            producer_prop_file += "\ndelivery.timeout.ms=%s\n" % (self.request_timeout_sec * 1000 * self.retries)
```

Comment:
The `retries` parameter is ignored if EOS is enabled.

Also, along with passing the `retries` config along to the producer, we set the delivery timeout high enough to accomodate the desired retry value.